libbindgen/src/codegen/mod.rs

```diff
@@ -656,8 +656,30 @@ impl CodeGenerator for CompInfo {
         //
         // TODO: Generate layout tests for template specializations, yay!
         if self.has_non_type_template_params() ||
-           self.is_template_specialization() {
-            return;
+            self.is_template_specialization() {
+                let layout = item.kind().expect_type().layout(ctx);
+                let canonical_name = item.canonical_name(ctx);
+
+                if let Some(layout) = layout {
+                    let fn_name = format!("bindgen_test_layout_template_{}", canonical_name);
```

Comment:
As discussed before, this will probably need to include the type id of the specialization, or similar, so it generates a totally unique name.